cinema/urls.py

```diff
@@ -2,9 +1,41 @@
+from django.urls import path, include
+from rest_framework import routers
 
-from cinema.views import movie_list, movie_detail
+from cinema.views import (
+    GenreList,
+    GenreDetail,
+    ActorList,
+    ActorDetail,
+    CinemaViewSet,
+    MovieViewSet
+)
+
+
+cinema_hall_list = CinemaViewSet.as_view(actions={
+    "get": "list",
+    "post": "create"
+})
+cinema_hall_detail = CinemaViewSet.as_view(actions={
+    "get": "retrieve",
+    "put": "update",
+    "patch": "partial_update",
+    "delete": "destroy"
+})
+
+router = routers.DefaultRouter()
+router.register("movies", MovieViewSet, basename="movies")
 
 urlpatterns = [
-    path("movies/", movie_list, name="movie-list"),
-    path("movies/<int:pk>/", movie_detail, name="movie-detail"),
+    path("genre/", GenreList.as_view(), name="movie-list"),
+    path("genre/<int:pk>/", GenreDetail.as_view(), name="movie-detail"),
+
+    path("actor/", ActorList.as_view(), name="movie-list"),
+    path("actor/<int:pk>/", ActorDetail.as_view(), name="movie-detail"),
```

Comment:
The name 'movie-detail' for the ActorDetail path is misleading. Consider renaming it to 'actor-detail' to accurately reflect the resource being accessed.